src/test/fuzz/coins_view.cpp

```diff
@@ -151,7 +151,9 @@ FUZZ_TARGET(coins_view, .init = initialize_coins_view)
                     coins_view_cache.BatchWrite(cursor, fuzzed_data_provider.ConsumeBool() ? ConsumeUInt256(fuzzed_data_provider) : coins_view_cache.GetBestBlock());
                     expected_code_path = true;
                 } catch (const std::logic_error& e) {
-                    if (e.what() == std::string{"FRESH flag misapplied to coin that exists in parent cache"}) {
+                    if (e.what() == std::string{"FRESH flag misapplied to coin that exists in parent cache"}
```

Comment:
I think after this change is merged, we can remove `SetFresh` and then the fuzz harness will not be able to add a FRESH-but-not-DIRTY. Then we can remove these catches and actually break if the fuzzer reaches them.

Right now the fuzz harness can still call `SetFresh` without calling `SetDirty`.

Reply:
The other 2 though I think you're right. They should not be possible to hit even in the fuzz harness now. Will remove and fuzz.

Reply:
Looking again, these can still be called with the fuzz harness. Removing `SetFresh` in a follow-up would allow us to remove one of them, but the other 2 would not since the `random_coin` can be a spent coin.

Reply:
> since the random_coin can be a spent coin

But can that happen in reality or are we testing something that's difficult to justify?